File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Effect/Combat/WebParticleEffect.ts
import { FairyGUI } from "csharp";
import { View_WebParticleEffect } from "../../Gen/View/ModTheSpire_Effect";
import { AbstractEffect } from "../AbstractEffect";


export class WebParticleEffect extends AbstractEffect{
    public Loader : FairyGUI.GComponent;
    public constructor(private _X:number,private _Y:number) {
        super();
        this.Loader = <FairyGUI.GComponent>View_WebParticleEffect.CreateInstance();
        this.Loader.GetTransition("t0").Play();
        this.Loader.SetPosition(_X,_Y,0);
    }
    
    public Update() {
        super.Update();
        if(this.Duration >= 1){
            this.Loader.Dispose();
            this.IsDone = true;
        }
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Combat/WebParticleEffect.cs
using FairyGUI;
using ModTheSpire.ModTheSpire_Effect;
using ModTheSpire.Unit;

namespace ModTheSpire.Effect.Combat
{
    public class WebParticleEffect : AbstractEffect
    {
        private float mX;
        private float mY;
        private GComponent mloader;
        public WebParticleEffect(float x, float y)
        {
            mX = x;
            mY = y;
            mloader = View_WebParticleEffect.CreateInstance();
            mloader.GetTransition("t0").Play();
            mloader.SetPosition(x,y,0);
        }

        public override void Update()
        {
            base.Update();
            if(Duration >= 1)
                Finish();
        }

        public override void Dispose()
        {
            base.Dispose();
            mloader.Dispose();
        }
    }
}